pkg/cmd/server/api/types.go

```diff
@@ -673,6 +673,17 @@ type GitHubIdentityProvider struct {
 	ClientSecret string
 }
 
+type GitLabIdentityProvider struct {
+	unversioned.TypeMeta
+
```

Comment:
add a field to hold a custom CA bundle (add add the ref to GetMasterFileReferences):
```
CA string `json:"ca"`
```